src/vit-servicing-station/vit-servicing-station-tests/src/common/raw_snapshot.rs

```diff
@@ -0,0 +1,125 @@
+use chain_impl_mockchain::testing::TestGen;
+use jormungandr_lib::interfaces::Value;
+use rand::Rng;
+use snapshot_lib::registration::{Delegations, VotingRegistration};
+use snapshot_lib::Fraction;
+use snapshot_lib::{
+    voting_group::{DEFAULT_DIRECT_VOTER_GROUP, DEFAULT_REPRESENTATIVE_GROUP},
+    CATALYST_VOTING_PURPOSE_TAG,
+};
+use time::OffsetDateTime;
+use vit_servicing_station_lib::v0::endpoints::snapshot::RawSnapshotInput;
+
+#[derive(Debug, Clone)]
+pub struct RawSnapshot {
+    pub tag: String,
+    pub content: RawSnapshotInput,
+}
+
+impl Default for RawSnapshot {
+    fn default() -> RawSnapshot {
+        RawSnapshotBuilder::default().build()
+    }
+}
+
+#[derive(Debug)]
+pub struct RawSnapshotBuilder {
+    tag: String,
+    update_timestamp: i64,
+    min_stake_threshold: Value,
+    voting_power_cap: Fraction,
+    direct_voters_group: Option<String>,
+    representatives_group: Option<String>,
+    voting_registrations_count: usize,
+}
+
+impl Default for RawSnapshotBuilder {
+    fn default() -> RawSnapshotBuilder {
+        Self {
+            tag: "daily".to_string(),
+            update_timestamp: OffsetDateTime::now_utc().unix_timestamp(),
+            min_stake_threshold: 0.into(),
+            voting_power_cap: 100.into(),
+            direct_voters_group: Some(DEFAULT_DIRECT_VOTER_GROUP.into()),
+            representatives_group: Some(DEFAULT_REPRESENTATIVE_GROUP.into()),
+            voting_registrations_count: 2,
+        }
+    }
+}
+
+impl RawSnapshotBuilder {
+    pub fn with_tag<S: Into<String>>(mut self, tag: S) -> Self {
+        self.tag = tag.into();
+        self
+    }
+
+    pub fn with_voting_registrations_count(mut self, voting_registrations_count: usize) -> Self {
+        self.voting_registrations_count = voting_registrations_count;
+        self
+    }
+
+    pub fn with_timestamp(mut self, timestamp: i64) -> Self {
+        self.update_timestamp = timestamp;
+        self
+    }
+
+    pub fn build(self) -> RawSnapshot {
+        let mut rng = rand::rngs::OsRng;
+        let mut delegation_type_count = 0;
+
+        RawSnapshot {
+            content: RawSnapshotInput {
+                snapshot: std::iter::from_fn(|| {
+                    Some(VotingRegistration {
```

Comment:
I think this is so useful it should be extracted as independent struct which responsibility is only to create random voting registration. Definitely see potenial to use it in other parts of tests

Reply:
done